lighty-modules/lighty-aaa-aggregator/lighty-aaa/src/test/java/io/lighty/aaa/ShiroInitializationTest.java

```diff
@@ -0,0 +1,111 @@
+import org.opendaylight.yang.gen.v1.config.aaa.authn.encrypt.service.config.rev160915.AaaEncryptServiceConfigBuilder;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.yang.aaa.cert.mdsal.rev160321.KeyStores;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.yang.aaa.cert.mdsal.rev160321.KeyStoresBuilder;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.yang.aaa.cert.mdsal.rev160321.key.stores.SslData;
+import org.opendaylight.yang.gen.v1.urn.opendaylight.yang.aaa.cert.mdsal.rev160321.key.stores.SslDataBuilder;
+import org.opendaylight.yangtools.util.concurrent.FluentFutures;
+import org.opendaylight.yangtools.yang.binding.InstanceIdentifier;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class ShiroInitializationTest {
+    private static final AAAConfiguration AAA_CONFIGURATION = AAAConfigUtils.createDefaultAAAConfiguration();
+    private static final String BUNDLE_NAME = "opendaylight";
+    @Mock
+    private LightyServerBuilder server;
+    @Mock
+    private DataBroker bindingDataBroker;
+    @Mock
+    private ReadTransaction readTransaction;
+    @Mock
+    private CredentialAuth<PasswordCredentials> credentialAuth;
+    private AAALighty aaaLighty;
+
+    @BeforeClass
+    public void init() {
+        // Initialize the mock objects
+        MockitoAnnotations.initMocks(this);
+
+        // Set up mocks for some datastore reads
+        when(bindingDataBroker.newReadOnlyTransaction()).thenReturn(readTransaction);
+        AaaEncryptServiceConfig aaaEncryptServiceConfig = new AaaEncryptServiceConfigBuilder().build();
+        when(readTransaction.read(LogicalDatastoreType.CONFIGURATION, MdsalUtils.getEncryptionSrvConfigIid()))
+                .thenReturn(FluentFutures.immediateFluentFuture(Optional.of(aaaEncryptServiceConfig)));
+
+        KeyStores keyStores = new KeyStoresBuilder().build();
+        when(readTransaction.read(LogicalDatastoreType.CONFIGURATION, InstanceIdentifier.create(KeyStores.class)))
+                .thenReturn(FluentFutures.immediateFluentFuture(Optional.of(keyStores)));
+
+        SslData sslData = new SslDataBuilder()
+                .setBundleName(BUNDLE_NAME).build();
+        when(readTransaction.read(LogicalDatastoreType.CONFIGURATION, KeyStoresDataUtils.getSslDataIid(BUNDLE_NAME)))
+                .thenReturn(FluentFutures.immediateFluentFuture(Optional.of(sslData)));
+    }
+
+    // Test that the AAALighty object can be stopped after failing to initialize
+    @Test
+    public void testStopProcedureWithFailedInitialization() {
+        // Create an AAALighty object with mocked dependencies
+        this.aaaLighty = new AAALighty(bindingDataBroker, credentialAuth, server, AAA_CONFIGURATION);
+        // Ensure that the object was created successfully
+        assertNotNull(aaaLighty);
+        // Expect an Exception to be thrown when trying to initialize the object
+        expectThrows(Exception.class, () -> aaaLighty.initProcedure());
```

Comment:
If the exception is not thrown, then aaaLighty.stopProcedure() will not be called. Therefore, I would recommend using:
```java
    @AfterMethod
    public void tearDown() {
        if (aaaLighty != null) {
            // Stop the object and ensure that stopping was successful
            assertTrue(aaaLighty.stopProcedure());
        }
    }
```

Reply:
done